gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -176,7 +176,12 @@ public D readRecordImpl(D reuse) throws DataRecordException, IOException {
 
           D record = decodeKafkaMessage(nextValidMessage);
 
-          this.statsTracker.onDecodeableRecord(this.currentPartitionIdx, readStartTime, decodeStartTime, nextValidMessage.getValueSizeInBytes());
+          long logAppendTime = 0L;
+          if (nextValidMessage.isTimestampLogAppend()) {
```

Comment:
If this means we only enable SLA when the record timestamp type is LogAppendTime. And when it's CreateTime we do not track the SLA?

Reply:
Yes. Record Create time is a little tricky since we have not control over record creation and SLA misses based on create time may result in false positives. Having said that, I think it would be useful to have another metric to track the average delay between record creation time and extract time. Can be a separate PR.